src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -3588,24 +3588,93 @@ public void LogLevelSerialization(LogLevel expectedLevel)
                 bool logLevelPropertyExists = telemetryElement.TryGetProperty("log-level", out JsonElement logLevelElement);
                 Assert.AreEqual(expected: true, actual: logLevelPropertyExists);
 
-                //Validate level property inside log-level is of expected value
-                bool levelPropertyExists = logLevelElement.TryGetProperty("level", out JsonElement levelElement);
-                Assert.AreEqual(expected: true, actual: levelPropertyExists);
+                //Validate the dictionary inside the log-level property is of expected value
+                bool dictionaryLogLevelExists = logLevelElement.TryGetProperty("default", out JsonElement levelElement);
+                Assert.AreEqual(expected: true, actual: dictionaryLogLevelExists);
                 Assert.AreEqual(expectedLevel.ToString().ToLower(), levelElement.GetString());
             }
         }
 
+        /// <summary>
+        /// Tests different log level filters that are valid and check that they are deserialized correctly
+        /// </summary>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(LoggerFilters.RUNTIME_CONFIG_VALIDATOR_FILTER)]
+        [DataRow(LoggerFilters.SQL_QUERY_ENGINE_FILTER)]
+        [DataRow(LoggerFilters.IQUERY_EXECUTOR_FILTER)]
+        [DataRow(LoggerFilters.ISQL_METADATA_PROVIDER_FILTER)]
+        [DataRow(LoggerFilters.BASIC_HEALTH_REPORT_RESPONSE_WRITER_FILTER)]
+        [DataRow(LoggerFilters.COMPREHENSIVE_HEALTH_REPORT_RESPONSE_WRITER_FILTER)]
+        [DataRow(LoggerFilters.REST_CONTROLLER_FILTER)]
+        [DataRow(LoggerFilters.CLIENT_ROLE_HEADER_AUTHENTICATION_MIDDLEWARE_FILTER)]
+        [DataRow(LoggerFilters.CONFIGURATION_CONTROLLER_FILTER)]
+        [DataRow(LoggerFilters.IAUTHORIZATION_HANDLER_FILTER)]
+        [DataRow(LoggerFilters.IAUTHORIZATION_RESOLVER_FILTER)]
+        [DataRow(LoggerFilters.DEFAULT_FILTER)]
+        public void ValidLogLevelFilters(string loggingFilter)
+        {
+            RuntimeConfig configWithCustomLogLevel = InitializeRuntimeWithLogLevel(LogLevel.Debug, loggingFilter);
+
+            string configWithCustomLogLevelJson = configWithCustomLogLevel.ToJson();
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(configWithCustomLogLevelJson, out RuntimeConfig deserializedRuntimeConfig));
+
+            SortedList<string, LogLevel?> actualLoggerLevel = deserializedRuntimeConfig.Runtime.Telemetry.LoggerLevel;
+            Assert.IsTrue(actualLoggerLevel.ContainsKey(loggingFilter) && actualLoggerLevel.Count == 1);
+        }
```

Comment:
Why are we not asserting the actual value of the log-level with the specified logger filter?